systests/spring-boot/src/test/java/org/apache/cxf/systest/jaxrs/spring/boot/SpringJaxrsObservabiityTest.java

```diff
@@ -0,0 +1,173 @@
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.hasSize;
+
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT, classes = SpringJaxrsObservabiityTest.TestConfig.class)
+@ActiveProfiles("jaxrs")
+@AutoConfigureObservability
+public class SpringJaxrsObservabiityTest {
+
+    @Autowired
+    private MeterRegistry registry;
+    
+    @Autowired
+    private ObservationRegistry observationRegistry;
+
+    @Autowired
+    private ArrayListSpanReporter arrayListSpanReporter;
+
+    @LocalServerPort
+    private int port;
+
+    @EnableAutoConfiguration(exclude = WebMvcObservationAutoConfiguration.class)
+    @ComponentScan(basePackageClasses = Library.class)
+    static class TestConfig {
+        @Bean
+        public Feature observationFeature(ObservationRegistry observationRegistry) {
+            return new ObservationFeature(observationRegistry);
+        }
+        
+        @Bean
+        public JacksonJsonProvider jacksonJsonProvider() {
+            return new JacksonJsonProvider();
+        }
+
+        @Bean
+        Sampler sampler() {
+            return Sampler.ALWAYS_SAMPLE;
+        }
+
+        @Bean
+        ArrayListSpanReporter arrayListSpanReporter() {
+            return new ArrayListSpanReporter();
+        }
+
+        @Bean
+        Reporter<Span> reporter() {
+            return Reporter.CONSOLE;
+        }
+    }
+
+    @Autowired
+    public void setBus(Bus bus) {
+        // By default, the exception are propagated and out fault interceptors are not called 
+        bus.setProperty("org.apache.cxf.propagate.exception", Boolean.FALSE);
+    }
+
+    @AfterEach
+    public void clear() {
+        registry.clear();
+        arrayListSpanReporter.close();
+    }
+
+    @Test
+    public void testJaxrsSuccessMetric() {
+        final WebTarget target = createWebTarget();
+        
+        try (Response r = target.request().get()) {
+            assertThat(r.getStatus()).isEqualTo(200);
+        }
+        
+        await()
+            .atMost(Duration.ofSeconds(1))
+            .until(() -> arrayListSpanReporter.getSpans(), hasSize(2));
+
+        // Micrometer Observation with Micrometer Tracing
+        SpansAssert.assertThat(arrayListSpanReporter.getSpans())
+               .haveSameTraceId()
+               .hasASpanWithName("GET", spanAssert -> {
+                   spanAssert.hasKindEqualTo(Kind.CLIENT)
+                             .hasTag("rpc.service", "WebClient")
+                             .hasTag("rpc.system", "cxf")
+                             .hasTagWithKey("server.address")
+                             .hasTagWithKey("server.port")
+                             .isStarted()
+                             .isEnded();
+               })
+               .hasASpanWithName("rpc.server.duration", spanAssert -> {
```

Comment:
The server span name is very unexpected `rpc.server.duration`, do you know why is that?

Reply:
Interesting. That would mean that the default, "metric" related name hooked in and the contextual name wasn't called. Here you can see how the contextual name should be set (https://github.com/apache/cxf/pull/1346/files#diff-05b29f9cca6ab07f827c4985ccf14e61f6c878510b4e18c37634b360041ed658R42) and here (https://github.com/apache/cxf/pull/1346/files#diff-3a451c2c9618eb5753ffd9960d384ffa6685f97fedf6605d787c7c018556d871R77) but maybe I messed sth up 😬

Reply:
I will try to debug that, but besides that - I think we should be all set

Reply:
Should be all set, the JAX-RS HTTP instrumentations works pretty well.